File: software/backend/src/main/java/ru/scms/back/dto/OrderDto.java
package ru.scms.back.dto;

import ru.scms.back.enums.OrderStatus;
import ru.scms.back.enums.SleeveGender;

import java.time.LocalDateTime;

public record OrderDto(
        Long id,
        String code,
        Long methUserId,
        String methUserName,
        Long sleeveId,
        String sleeveCode,
        SleeveGender gender,
        Integer height,
        Integer weight,
        Integer age,
        Long geneticArchiveId,
        String geneticArchiveName,
        OrderStatus status,
        LocalDateTime createdAt
) {
}

File: software/client-portal/src/shared/api/dto/orderDto.ts
import type { OrderStatus, SleeveGender } from "./enums";

export interface OrderDto {
  id: number;
  code: string;
  methUserId: number;
  methUserName: string | null;
  sleeveId: number | null;
  sleeveCode: string | null;
  gender: SleeveGender | null;
  height: number | null;
  weight: number | null;
  age: number | null;
  geneticArchiveId: number | null;
  geneticArchiveName: string | null;
  status: OrderStatus;
  createdAt: string;
}
